### PR TITLE
Rwc2022

### PR DESCRIPTION
**Pull request by PetervDooren**

### CODE REVIEW

File: launch/hardware/sensors/depth_to_laser.launch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+
```
Comment (MatthijsBurgh) [inline launch/hardware/sensors/depth_to_laser.launch]:
```suggestion
```

File: launch/hardware/sensors/depth_to_laser.launch [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+<launch>
+
+    <!-- Get the machine file -->
+    <arg name="machine" default="localhost"/>
+    <include file="$(env ROBOT_BRINGUP_PATH)/machines/$(arg machine).machine" />
+
+    <arg name="robot_name" />
+    <arg name="camera_info_depth" default="/$(arg robot_name)/head_rgbd_sensor/depth_registered/camera_info" />
+    <arg name="image_depth" default="/$(arg robot_name)/head_rgbd_sensor/depth_registered/image" />
+    <arg name="scan_projected" default="/$(arg robot_name)/head_rgbd_sensor/scan" />
```
Comment (MatthijsBurgh) [inline launch/hardware/sensors/depth_to_laser.launch]: We always run this in NS in the parent launch file. So no need for this robot NS prefix.

File: launch/hardware/sensors/depth_to_laser.launch [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    <node pkg="depthimage_to_laserscan" type="depthimage_to_laserscan" name="depthimage_to_laserscan" output="screen" machine="$(arg machine)">
+        <remap from="image" to="$(arg image_depth)" />
+        <remap from="camera_info" to="$(arg camera_info_depth)" />
+        <remap from="scan" to="$(arg scan_projected)"/>
+        <rosparam command="load" file="$(env ROBOT_BRINGUP_PATH)/parameters/hardware/sensors/head_camera_laser_projection.yaml" />
```
Comment (MatthijsBurgh) [inline launch/hardware/sensors/depth_to_laser.launch:21]: Might want to make this configurable as well

File: launch/hardware/sensors/depth_to_laser.launch [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    </node>
+
+    <node pkg="tf2_ros" type="static_transform_publisher" name="head_camera_laser_broadcaster" args="0 0 0 0 -1.57 1.57 head_rgbd_sensor_rgb_frame head_rgbd_sensor_laser_frame" />
```
Comment (MatthijsBurgh) [inline launch/hardware/sensors/depth_to_laser.launch]: This is not generic at all. So should be remove or made generic/dynamic.

File: launch/hardware/sensors/depth_to_laser.launch [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
```
Comment (MatthijsBurgh) [inline launch/hardware/sensors/depth_to_laser.launch]:
```suggestion
```

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: launch/hardware/sensors/depth_to_laser.launch (hunk 3 above)
Comment (PetervDooren) [reply launch/hardware/sensors/depth_to_laser.launch:21]: This is exactly like the other launch files. No need to change this

**Review by MatthijsBurgh — APPROVED** [review]
@PetervDooren I also made the transform configurable

**Comment by PetervDooren** [comment]
As mentioned before I dont think [8cab438](https://github.com/tue-robotics/robot_launch_files/pull/72/commits/8cab43854771cde8445b0be4c9e9219bec84215a) will ever be used

**Pull request merged**